src/pages/index.js

```diff
@@ -0,0 +1,299 @@
+  }
+}
+
+const openModal = (modal) => {
+  modal.classList.add("modal_is-opened");
+  document.addEventListener("keydown", handleEscKey);
+  modal.addEventListener("click", handleOverlayClick);
+};
+
+const closeModal = (modal) => {
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleEscKey);
+  modal.removeEventListener("click", handleOverlayClick);
+};
+
+const addOpenModalListener = (trigger, modal, cb) => {
+  trigger.addEventListener("click", () => {
+    if (cb) cb();
+    openModal(modal);
+  });
+};
+
+const addCloseModalListener = (button, modal) => {
+  button.addEventListener("click", () => closeModal(modal));
+};
+
+function handleLikeClick(likeBtn, cardId) {
+  const isLiked = likeBtn.classList.contains("card__like-btn_active");
+  api
+    .changeLikeStatus(cardId, !isLiked)
+    .then((card) => {
+      const isNowLiked = card.isLiked;
+      likeBtn.classList.toggle("card__like-btn_active", isNowLiked);
+    })
+    .catch((err) => console.error("Error changing like status:", err));
+}
+
+function getCardElement(data) {
+  const cardElement = dom.cardTemplate.cloneNode(true);
+  cardElement.querySelector(".card__title").textContent = data.name;
+  const cardImage = cardElement.querySelector(".card__image");
+  cardImage.alt = data.name;
+  cardImage.src = data.link;
+  const likeBtn = cardElement.querySelector(".card__like-btn");
+  const isLiked = (data.likes || []).some((user) => user._id === currentUserId);
+  likeBtn.classList.toggle("card__like-btn_active", isLiked);
+  if (data.isLiked) {
+    likeBtn.classList.add("card__like-btn_active");
+  } else {
+    likeBtn.classList.remove("card__like-btn_active");
+  }
+  likeBtn.addEventListener("click", () => handleLikeClick(likeBtn, data._id));
+  cardElement.querySelector(".card__del-btn").addEventListener("click", () => handleDelCard(cardElement, data));
+  cardImage.addEventListener("click", () => openPreviewModal(data));
+  return cardElement;
+}
+
+function renderCard(data, container) {
+  const cardElement = getCardElement(data);
+  container.prepend(cardElement);
+}
+
+function openPreviewModal(data) {
+  dom.previewImage.src = data.link;
+  dom.previewImage.alt = data.name;
+  dom.previewCaption.textContent = data.name;
+  openModal(dom.previewModal);
+}
+
+function handleDelCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(dom.deletePostModal);
+}
+
+function handleDelSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(dom.submitDeletePostBtn, "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(dom.deletePostModal);
+      selectedCard = null;
+      selectedCardId = null;
+    })
+    .catch((err) => console.error("Error deleting card:", err))
+    .finally(() => {
+      setButtonText(dom.submitDeletePostBtn, "Delete");
+      disableBtn(dom.submitDeletePostBtn, settings);
```

Comment:
* I can't delete 2 cards in a row because the submit button gets **disabled** after the 1st confirmation

Please, don't disable the button here